### PR TITLE
Add an EDN parameter doc emitter

### PR DESCRIPTION
**Pull request by WickedShell**
This is the result of poking around a bit because I'd find EDN more convenient then XML for parsing. Depends upon `edn_format` being installed from pip.

Also removes the `,`'s from the FLIGHT_OPTIONS on plane as it breaks the other parsers as well (IE the wiki looks bad http://ardupilot.org/plane/docs/parameters.html#flight-options-flight-mode-options )

For those familiar with EDN the structure this uses is:
``` clojure
{:date #inst "2018-12-06T08:32:34.091777Z" ; utc time the parameters were parsed
 :git-hash "14fc764a17"
 :vehicle "ArduPlane" ; vehicle name this parameter set maps out to
 "FORMAT_VERSION" {:name "FORMAT_VERSION"
                   :description "This value is incremented when changes are made to the eeprom format"
                   :user "Advanced"
                   :display-name "Eeprom format version number"}
 ; rest of parameters
}
```

In the map for each parameter `:bitmask` will contain a vector of the descriptions per bit (with a null if the bit has no known meaning), `:values` is a map of the floating point value to it's description, `:increment` is a floating point number, `:range` is a map containing a low and high key to be used for creating the range.

After more suffering this runs under both python 2 and 3.

### CODE REVIEW

File: Tools/autotest/param_metadata/param_parse.py [hunk 1]
```diff
@@ -335,5 +335,10 @@ def do_emit(emit):
     do_emit(HtmlEmit())
     do_emit(RSTEmit())
     do_emit(MDEmit())
+    try:
+        from ednemit import EDNEmit
+        do_emit(EDNEmit())
+    except:
+        print("Unable to emit EDN, install edn_format if desired")
```
Comment (peterbarker) [inline Tools/autotest/param_metadata/param_parse.py]: Exception would have to be tighter to justify this error message.

Comment (peterbarker) [inline Tools/autotest/param_metadata/param_parse.py]: Please wrap this in a check to `opts.verbose` - it will avoid support calls.

Comment (WickedShell) [reply Tools/autotest/param_metadata/param_parse.py]: Tightened to just `ImportError` is that fine?

**Comment by WickedShell** [comment]
I scope crept this on myself slightly, `param_parse.py` now accepts a `--format` command line option that specifies what emitter(s) to use. I was getting rather annoyed at creating a bunch of junk in my ArduPilot folder that wasn't even in gitignore (added them to the ignore list as well), and really only want the one file anyways.

**Pull request merged**